projects/vdk-core/src/vdk/internal/builtin_plugins/ingestion/ingester_router.py

```diff
@@ -41,6 +42,16 @@ def __init__(self, cfg: Configuration, core_state: StateStore):
         self._cached_ingesters: Dict[str, IngesterBase] = dict()
         self._ingester_builders: Dict[str, IngesterPluginFactory] = dict()
 
+        self._pre_processor_sequence: Optional[List] = self.__get_plugin_sequence(
+            "INGEST_PAYLOAD_PREPROCESS_SEQUENCE"
+        )
+        self._post_processor_sequence: Optional[List] = self.__get_plugin_sequence(
+            "INGEST_PAYLOAD_POSTPROCESS_SEQUENCE"
+        )
+
+        self._initialized_pre_processors: List[Optional[IIngesterPlugin]] = []
+        self._initialized_post_processors: List[Optional[IIngesterPlugin]] = []
```

Comment:
I don't understand why adding four fields;
`self._ingester_builders` are populated with all ingested plugins already, what is needed additionally,
is literally "give me pre-processors out of those, then filter by config-provided pre-sequence",
also "give me post-processors out of those, then filter by config-provided post-sequence".

I would consider this a short lambda potentially. And even name of pre_ingest_process/post_ingest_process along with INGEST_PAYLOAD_PREPROCESS_SEQUENCE/INGEST_PAYLOAD_POSTPROCESS_SEQUENCE could be parameterized, so one lambda is needed for both similar cases.

Then, when evaluating the processors 
-> common to ingestion itself, where method configured is missing raise misconfiguration exception
-> if number of filtered pre-processors differs than config.split then raise exception
-> if number of filtered post-...same

Reply:
Removed unnecessary fields, and did some refactoring.